packages/ui/src/common/providers/network-endpoints/provider.tsx

```diff
@@ -38,14 +38,20 @@ export const NetworkEndpointsProvider = ({ children }: Props) => {
           nodeRpcEndpoint: config['websocket_rpc'],
         }
 
-        const shouldUpdateConfig = !objectEquals<Partial<NetworkEndpoints>>(newNetworkConfig)(storedNetworkConfig ?? {})
-        const shouldUpdateNetwork = network !== 'olympia-testnet'
+        if (!endpointsAreDefined(newNetworkConfig)) {
+          throw new Error('fetched config missing endpoints')
+        }
```

Comment:
This will get caught by the `catch` on line 66 and throw the wrong error. But I shouldn't have put this catch statement this far anyway it should just be to line 33 :smile:

Reply:
Yes I was in-fact relying on that catch block, and had originally changed the throw in it to include the `${err}`.
But your suggestion is better. Will fix.

Reply:
done in [7e017ba](https://github.com/Joystream/pioneer/pull/2532/commits/7e017baf77ef11cac128aa61725bda531b2bfa9b)